app/cdap/components/ResourceCenterEntity/PullPipelineWizard.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import Alert from 'components/shared/Alert';
+import React from 'react';
+import { Provider } from 'react-redux';
+import SourceControlManagementSyncStore from 'components/SourceControlManagement/store';
+import { RemotePipelineListView } from 'components/SourceControlManagement/RemotePipelineListView';
+import StandardModal from 'components/shared/StandardModal';
+import T from 'i18n-react';
+
+export const reducer = (state, action) => {
```

Comment:
@seanfeld I didn't want to create another reducer file for something this simple so i just exported here. Let me know if this is legal

Reply:
yeah this is fine

Reply:
wait... what is going on here? this component is only used in one spot?

Reply:
This whole flow is kinda weird but at the bare minimum since you're doing it like this you should look up how to type the dispatch correctly instead of (obj: object) => void

Reply:
actually, is it better to export the state and dispatch function of useReducer, rather than the reducer itself? so that I don't need to pass the dispatch function anymore